snowshu/adapters/source_adapters/snowflake_adapter.py

```diff
@@ -348,6 +348,7 @@ def predicate_constraint_statement(
                 f"FROM ("
                 f"    SELECT DISTINCT {remote_key} "
                 f"    FROM {relation.temp_dot_notation} "
+                f"    WHERE {remote_key} NOT LIKE '%''%' "
```

Comment:
While testing fixed solution, I encountered an edge case where value in returned list contained single quote eg. **kacper.w'ozniak@mail.com** - this was causing failing sample table creation, hence I added a heuristic to avoid values containing single quote. (It's hard to replicate the issue as every time it's sampling a different subset)

Reply:
Why don't we try escaping quotes in the case of string values, instead of excluding them?

Reply:
I was thinking that the amount of such edge cases is minor and if we escape it and then later try to use it again to filter, it may not match the pattern or cause some other downstream issues, hence excluding this minor number of records containing quotes seemed to be simpler and safer in my opinion.

Reply:
Since we are implementing this on the Snowflake side, it might make more sense to pass a query instead of a list of keys, what do you think?